p8e-api/src/main/kotlin/io/provenance/engine/service/JobService.kt

```diff
@@ -0,0 +1,55 @@
+package io.provenance.engine.service
+
+import com.google.protobuf.Message
+import io.p8e.util.ThreadPoolFactory
+import io.p8e.util.toHex
+import io.p8e.util.toPublicKey
+import io.provenance.p8e.shared.domain.JobRecord
+import io.provenance.p8e.shared.domain.JobStatus
+import io.provenance.p8e.shared.extension.logger
+import org.jetbrains.exposed.sql.transactions.transaction
+import org.springframework.scheduling.annotation.Scheduled
+import org.springframework.stereotype.Component
+import p8e.Jobs
+import java.lang.IllegalArgumentException
+import java.util.concurrent.CompletableFuture
+import kotlin.concurrent.thread
+
+@Component
+class JobService(private val jobHandlerServiceFactory: JobHandlerServiceFactory) {
```

Comment:
what is the job service actually doing?

Reply:
polling for jobs and then dispatching them to the appropriate handlers (kind of like event retry, but a bit different)